### ISSUE TITLE
[Question] Wiki Edits

### ISSUE THREAD

**Issue by Mr00Anderson** [junkdog/artemis-odb#579]
Is there a general process you want for updating the wiki?

I updated the serialization page. I know their are multiple ways of going it, but the methods listed did not work.

**Comment by junkdog** [comment]
> Is there a general process you want for updating the wiki?

If you feel something is wrong, missing or otherwise in need of improvement - go ahead and edit.


> I updated the serialization page. I know their are multiple ways of going it, but the methods listed did not work.

Looking good contentwise, but github really likes cutting off code at the right margin. Maybe introduce a couple of linebreaks, alt shorten some variable names or dropping modifiers?

I guess we ought to consolidate all serialization examples - json, libgdx-json and kryo - under one page; right now https://github.com/junkdog/artemis-odb/wiki/libgdx-json and https://github.com/junkdog/artemis-odb/wiki/other-JSON deviate somewhat, even though the API is the same.

**Comment by Mr00Anderson** [comment]
Json wikis would benefit from being merged. I think the plugin move with issue #467 can help that consolidation some. 

The Wiki does feel spread out and broken up that its can become complicated to page back and fourth. Seems a lot to maintain for being agile as you and @DaanVanYperen.. (I've been thinking how to consolidate it and or have a single project example). I wanna help connect your library to others as an alternate to using any a game engine for prototyping and general development through the wiki and a world editor with ImGui  from JImGui.

**Comment by Mr00Anderson** [comment]
@junkdog I just learned that sometimes using the Java tags for MD causes the no scroll bars.

**Comment by DaanVanYperen** [comment]
What timing! Was evaluating the wiki yesterday, sketching out some thoughts in gdoc. 

> The Wiki does feel spread out and broken up that its can become complicated to page back and fourth. 

What's missing most is structure and flow, and tailoring to user groups. Quality wise most of the content is concise and to the point, but it's spread out all over the place. Can be improved in consistency, some meandering pages could do with a makeover, and probably fill in some gaps.

> I've been thinking how to consolidate it and or have a single project example

Discord is filled with Ashley users who are intimidated by odb, while core odb is equal in complexity to Ashley. I believe the need to go kitchen-sink is part of the problem (libgdx-artemis-quickstart). 

It's one of the reasons https://github.com/DaanVanYperen/artemis-odb-examples is now a thing. 

There's plenty we can do to make information easier to find and less overwhelming.

> Seems a lot to maintain for being agile

I believe @junkdog is fairly happy about the scope of odb and right now we don't have big pivots planned. Maintenance isn't to bad given that and users are fairly happy. Think we'd both probably go for green fielding something before turning very drastically (Kotlin/other web backends/etc).

I agree odb is a bit bloated (3 impls for serialization in core, but also several ways to alter entities), and I think taking a step back and sketching a horizon might be worth the time. Odb's extension points have proven themselves, and we could reasonably push towards a more modular design. Archetypes, pooling, managed links, fluid entities, as plugins etc. That would improve our extension points even further.

I wouldn't want to get rid of fluid entities, but it could use a transformation. Something worth discussing.

> Is there a chat you hang out in? 

@junkdog hides from me during manic contrib phases. ;) For his sanity. I've been trying to get him on libgdx discord.

**Comment by Mr00Anderson** [comment]
Oh its longer. I try to not chat. Get excited, you know. 

> What's missing most is structure and flow, and tailoring to user groups. Quality wise most of the content is concise and to the point, but it's spread out all over the place. Can be improved in consistency, some meandering pages could do with a makeover, and probably fill in some gaps.`

Agreed. I will help makeover starting now as I use the pages. 

> Discord is filled with Ashley users who are intimidated by odb, while core odb is equal in complexity to Ashley. I believe the need to go kitchen-sink is part of the problem (libgdx-artemis-quickstart).

I think having a full sample no matter the engine would help. I will make an extension to Artemis for this reason public, a second sample of examples and a editor. I just need to peal it out of my game lib. It will be messy at first, still testing the designs in practice. The code runs as an OpenGL app right now. That's what it is and will stay, independent. Though you can run JImGui and LibGDX! I will wait to make artifacts though for the library. When the artifact time come I got agents on each OS here at home, I use DevOps Azure, but am familiar with others. I will work on making this public over now-weekend. Can be found here [jimgui-artemis-odb-world-editor](https://github.com/Mr00Anderson/jimgui-artemis-odb-world-editor) Also I do not dislike LibGDX, I use it. Just wanted to have my game data design-able without it. 

> I believe @junkdog is fairly happy about the scope of odb and right now we don't have big pivots planned. Maintenance isn't too bad given that and users are fairly happy. Think we'd both probably go for green fielding something before turning very drastically (Kotlin/other web backends/etc).

>I agree odb is a bit bloated (3 impls for serialization in core, but also several ways to alter entities), and I think taking a step back and sketching a horizon might be worth the time. Odb's extension points have proven themselves, and we could reasonably push towards a more modular design. Archetypes, pooling, managed links, fluid entities, as plugins etc. That would improve our extension points even further.

>I wouldn't want to get rid of fluid entities, but it could use a transformation. Something worth discussing.

I think its just right. I think it can be done in a way that only one serialization lib is needed. I need time to think about this, to much clouding my mind (just woke up). I will work on this this week. I need to understand the libs here more to start writing my own serialization code. Sketching the horizon would be worth it, can add the json to that. This would also help solidify anything in question, documentation will be done in the same process and benefit from not having to have major changes to it. 

My background:
If your wondering oh no here is another "editor" guy. I recently stopped working in Cybersecurity field after 13 years. I am working on this editor full time from home for my game I started this journey 3 years or so ago. Mostly complete computer science degree, should be done 2020. I am working backwards, haha. I am 35, got plenty of time. Also dislike Unity and Unreal. The amount of head crushing that goes on inside as a programmer on those engines. I really like to know how the stuff works under the hood.

**Comment by Mr00Anderson** [comment]
I committed the code I have so far to the library I linked above. It runs and opens a reflective component reader. Its messy as well from testing but I was using ImGui-Kotlin before I switched to JImGui. Needless to say super early stages of work. I have a undocumented plan and it will come together rover the next month or so. I will now use the public repo though and make regular working commits as I progress. Just need to slap a licences on.
